verticapy/jupyter/extensions/sql_magic.py

```diff
@@ -57,19 +57,606 @@ def sql_magic(
+        class_fare
+
+    .. ipython:: python
+        :suppress:
+
+        import verticapy.stats as st
+        class_fare = titanic_clean.groupby("pclass",
+                                   [st.avg(titanic_clean["fare"])._as("avg_fare")])
+        html_file = open("figures/jupyter_extensions_sql_magic_sql_magic_6.html", "w")
+        html_file.write(class_fare._repr_html_())
+        html_file.close()
+
+    .. raw:: html
+        :file: SPHINX_DIRECTORY/figures/jupyter_extensions_sql_magic_sql_magic_6.html
+
+    Use the 'class_fare' variable in a SQL query:
+
+    .. code-block:: python
+
+        %%sql
+        SELECT
+            x.*,
+            y.avg_fare
+        FROM titanic AS x LEFT JOIN (SELECT * FROM :class_fare) AS y
+        ON x.pclass = y.pclass;
+
+    **Execution**: 0.011s
+
+    .. ipython:: python
+        :suppress:
+
+        %%sql
+        SELECT
+            x.*,
+            y.avg_fare
+        FROM titanic AS x LEFT JOIN (SELECT * FROM :class_fare) AS y
+        ON x.pclass = y.pclass;
+
+    .. ipython:: python
+        :suppress:
+
+        titanic_class = _
+        html_file = open("figures/jupyter_extensions_sql_magic_sql_magic_7.html", "w")
+        html_file.write(titanic_class._repr_html_())
+        html_file.close()
+
+    .. raw:: html
+        :file: SPHINX_DIRECTORY/figures/jupyter_extensions_sql_magic_sql_magic_7.html
+
+    You can do the same with a tablesample:
+
+    .. code-block:: python
+
+        tb = {"name": ["Badr", "Arash"], "specialty": ["Python", "C++"]}
+        tb = vp.tablesample(tb)
+
+    .. code-block:: python
+
+        %%sql
+        SELECT * FROM :tb;
+
+    **Execution**: 0.014s
+
+    .. ipython:: python
+        :suppress:
+
+        tb = {"name": ["Badr", "Arash"], "specialty": ["Python", "C++"]}
+        tb = vp.TableSample(tb)
+
+    .. ipython:: python
+        :suppress:
+
+        %%sql
+        SELECT * FROM :tb;
+
+    .. ipython:: python
+        :suppress:
+
+        tb_test = _
+        html_file = open("figures/jupyter_extensions_sql_magic_sql_magic_8.html", "w")
+        html_file.write(tb_test._repr_html_())
+        html_file.close()
+
+    .. raw:: html
+        :file: SPHINX_DIRECTORY/figures/jupyter_extensions_sql_magic_sql_magic_8.html
+
+    And with a pandas.DataFrame:
+
+    .. ipython:: python
+
+        titanic_pandas = titanic.to_pandas()
+        titanic_pandas
+
+    .. code-block:: python
+
+        %%sql
+        SELECT * FROM :titanic_pandas;
+
+    .. ipython:: python
+        :suppress:
+
+        %%sql
+        SELECT * FROM :titanic_pandas;
+
+    .. ipython:: python
+        :suppress:
+
+        pandas_test = _
+        html_file = open("figures/jupyter_extensions_sql_magic_sql_magic_9.html", "w")
+        html_file.write(pandas_test._repr_html_())
+        html_file.close()
+
+    .. raw:: html
+        :file: SPHINX_DIRECTORY/figures/jupyter_extensions_sql_magic_sql_magic_9.html
+
```

Comment:
.. note:: VerticaPy will store the object in a temporary local table before executing the overall query, which facilitates integration with in-memory objects.